components/core/src/clp/Grep.cpp

```diff
@@ -1063,4 +1011,437 @@ size_t Grep::search(Query const& query, size_t limit, Archive& archive, File& co
 
     return num_matches;
 }
+
+void Grep::generate_query_substring_logtypes(
+        string& processed_search_string,
+        ByteLexer& lexer,
+        vector<std::set<QueryLogtype>>& query_substr_logtypes
+) {
+    // We need to differentiate between literal '*'/'?' and wildcards
```

Comment:
Instead of also tracking `cancels` (escape characters), we can simply preprocess `processed_search_string` to remove escape characters entirely and keep track of the positions of greedy and non-greedy wildcards, right? I think that would also simplify the logic below. E.g., one of the things we discussed offline was that we don't actually need to check for unnecessary escapes since `processed_search_string` has already had them removed.

Reply:
We need to skip processing escape characters in both the generate_query_substring_logtypes() loop and get_substring_variable_types() loop. So instead of removing tracking escape characters, we can just avoid repeating the escape logic by using it at these locations. 

Also, after more thought, I don't know if we can actually avoid tracking it as we originally discussed. In the logic for determining proceeding delimiters, even if the search string doesn't contain invalid escape characters, we still need to do the same checks. For example, "* word test\\test\* *". Lets say a '\' is not a delimiter and '*" is a delimiter. Then for "word" we need to check lexer.is_delimiter(processed_search_string[end_idx]), but for "test\\test" we need to check lexer.is_delimiter(processed_search_string[end_idx + 1]).

Reply:
Removed duplicate escape logic by reusing escape. Added explanations for logic using escape to make it clear why its needed and what is being done.